installer_osx/make_installer.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+# preflight check
+
+if [[ ! -f ./make_installer.sh ]]; then
+	echo "You must run this script from within the installer_osx directory!"
+	exit 1
+fi
+
+# version and locations
+
+VERSION="$1"
+VST2="../products/Surge.vst"
+VST3="../products/Surge.vst3"
+AU="../products/Surge.component"
+RSRCS="../resources/data"
+
+if [[ $1 == "" ]]; then
+	echo "You must specify the version you are packaging!"
+	echo "eg: ./make_installer.sh 1.0.6b4"
+	exit 1
+fi
+
+# try to build VST2 package
+
+if [[ -d $VST2 ]]; then
+	pkgbuild --analyze --root "$VST2" Surge_VST2.plist
+	pkgbuild --root "$VST2" --component-plist Surge_VST2.plist --identifier "com.vemberaudio.vst2.pkg" --version $VERSION --install-location "/Library/Audio/Plug-Ins/VST/Surge.vst" Surge_VST2.pkg
+	rm Surge_VST2.plist
+fi
+
+# try to build VST3 package
+
+if [[ -d $VST3 ]]; then
+	pkgbuild --analyze --root "$VST3" Surge_VST3.plist
+	pkgbuild --root "$VST3" --component-plist Surge_VST3.plist --identifier "com.vemberaudio.vst3.pkg" --version $VERSION --install-location "/Library/Audio/Plug-Ins/VST3/Surge.vst3" Surge_VST3.pkg
+	rm Surge_VST3.plist
+fi
+
+# try to build AU package
+
+if [[ -d $AU ]]; then
+	pkgbuild --analyze --root "$AU" Surge_AU.plist
+	pkgbuild --root "$AU" --component-plist Surge_AU.plist --identifier "com.vemberaudio.au.pkg" --version $VERSION --install-location "/Library/Audio/Plug-Ins/Components/Surge.component" Surge_AU.pkg
+	rm Surge_AU.plist
+fi
+
+# build resources package
+
+pkgbuild --analyze --root "$RSRCS" Surge_Resources.plist
+pkgbuild --root "$RSRCS" --component-plist Surge_Resources.plist --identifier "com.vemberaudio.resources.pkg" --version $VERSION --scripts ResourcesPackageScript --install-location "/tmp/Surge" Surge_Resources.pkg
+rm Surge_Resources.plist
+
+# create distribution.xml
+
+echo '<?xml version="1.0" encoding="utf-8"?>' > distribution.xml
```

Comment:
just a suggestion here, maybe use a heredoc to make this more readable? e.g.:

```xml
cat << EOF >> distribution.xml
<?xml version="1.0" encoding="utf-8"?>
<installer-gui-script minSpecVersion="1">
  <title>Install Surge $VERSION</title>
  <license file="License.txt" />
  <pkg-ref id="$PKGREF"/>
...
```

And just use the `if` blocks to set up the necessary variables

Reply:
hehe, yeah I already thought that after looking at the mess I created :)  there's a comment on the PR itself